.github/workflows/release.yaml

```diff
@@ -16,7 +16,55 @@ jobs:
     name: Test, lint, & build
     uses: ./.github/workflows/test.yaml
 
+  prerelease:
+    name: Update documentation
+    runs-on: ubuntu-latest
+    outputs:
+      pr_number: ${{ steps.create_pr.outputs.pr_number }}
+    steps:
+      - name: Generate token
+        id: generate_token
+        uses: tibdex/github-app-token@v2
+        with:
+          app_id: ${{ secrets.OS_GITHUB_APP_ID }}
+          private_key: ${{ secrets.OS_GITHUB_APP_PRIVATE_KEY }}
+
+      - name: "☁️ checkout repository"
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          token: ${{ steps.generate_token.outputs.token }}
+
+      - name: "🤲 Setup Just"
+        uses: extractions/setup-just@v2
+
+      - name: "📗 generate documentation"
+        run: just gen-docs
+
+      - name: Create PR for documentation update
+        id: create_pr
+        env:
+          GH_TOKEN: ${{ steps.generate_token.outputs.token }}
+        run: |
+          timestamp=$(date +"%Y%m%d%H%M%S")
+          branch_name="chore-automated-doc-updates-$timestamp"
+          git config --local user.email "action@github.com"
+          git config --local user.name "GitHub Action"
+          git checkout -b "$branch_name"
+          git add .
+          if git diff --staged --quiet; then
+            echo "No changes to commit"
+            echo "pr_number=" >> $GITHUB_OUTPUT
+          else
+            git commit -m "chore: update documentation ($timestamp)"
+            git push -u origin "$branch_name"
+            pr_number=$(gh pr create --title "chore: update documentation ($timestamp)" --body "Automated documentation update for release" --base ${{ github.ref_name }} --head "$branch_name")
+            echo "pr_number=$pr_number" >> $GITHUB_OUTPUT
+            gh pr merge $pr_number --auto --delete-branch --squash
+          fi
+
   release:
+    needs: [test, prerelease]
```

Comment:
This seems like a good approach, but would love feedback. Adding it as a step without a `needs` means the PR might not be merged for the docs yet before a release happens, and it can also cause an infinite loop of a release.

I put tests in here as well.

Reply:
I'd honestly go even simpler: we already automatically commit the `CHANGELOG.md` in the release action. So I'd say this could just be an additional step before the semantic-release flow (although, I think semantic-release fails if the git repo is dirty so it'd probably have to be a separate, quick followup commit).

So instead of creating a whole PR, I'd just commit it directly to the target release branch with `skip-ci` like we do with other release assets.

Although I could see the value in getting eyes on the generated docs via PR before merging them in. It could be easy to let the process get away from us and we never actually look at what's being generated.

Reply:
I wasn't sure if we wanted to commit and push, so sounds good!